runtime/dart_snapshot.cc

```diff
@@ -102,7 +96,7 @@ static std::shared_ptr<const fml::Mapping> ResolveVMData(
       settings.vm_snapshot_data,          // embedder_mapping_callback
       settings.vm_snapshot_data_path,     // file_path
       settings.application_library_path,  // native_library_path
-      DartSnapshot::kVMDataSymbol,        // native_library_symbol_name
+      kVmSnapshotDataCSymbol,             // native_library_symbol_name
```

Comment:
The `dart_api.h` versions of these constants on non-Apple platforms contain a leading underscore.  Does this change require changing other parts of the build process (e.g. `lib/snapshot/BUILD.gn`)?

Reply:
[`SymbolMapping`](https://github.com/flutter/engine/blob/e5209202fb0022a9a1cd1c0edb0fa67fdcd7ed11/fml/mapping.cc#L113) should account for that in this fallback and try both.

Reply:
SymbolMapping will try adding an underscore prefix.  To handle this, it would need to be extended to also try removing the underscore prefix.